src/RA_StringUtils.h

```diff
@@ -45,9 +45,21 @@ _NODISCARD inline auto StringPrintf(_In_z_ _Printf_format_string_ const CharT* c
     }
     else if constexpr(std::is_same_v<CharT, wchar_t>)
     {
+        // starting capacity to reduce the need to reallocate, it could still be too small but is handled below
+        sFormatted.reserve(64U);
+        nNeeded = -1; // assume it already failed (usually does)
         assert(std::wstring_view{ sFormat }.find(L"%n") == std::wstring_view::npos);
         assert(sFormatted.capacity() > 0U && (sFormatted.capacity() < RSIZE_MAX/sizeof(wchar_t)));
-        nNeeded = std::vswprintf(sFormatted.data(), sFormatted.capacity(), sFormat, pArgs);
+
+        while(nNeeded < 0)
+        {
+            nNeeded = std::vswprintf(sFormatted.data(), sFormatted.capacity(), sFormat, pArgs);
+            if (nNeeded < 0)
+            {
+                const auto nCap = sFormatted.capacity();
+                sFormatted.reserve(nCap*2); // if it's still too small
+            }
```

Comment:
Cross-contamination is holding back your PRs. Let's get this sorted out correctly in one and leave it out of the others.